neqo-transport/src/fc.rs

```diff
@@ -343,10 +366,42 @@ impl ReceiverFlowControl<StreamId> {
         builder: &mut PacketBuilder,
         tokens: &mut Vec<RecoveryToken>,
         stats: &mut FrameStats,
+        now: Instant,
+        rtt: Duration,
     ) {
         if !self.frame_needed() {
             return;
         }
+
+        // Auto-tune max_active, i.e. the flow control window.
+        //
+        // If the sending rate ( window_bytes used / elapsed ) exceeds the rate
+        // allowed by the maximum flow control window and the current rtt (
+        // max_active / rtt ), try to increase the maximum flow control window (
+        // max_active ).
+        if let Some(max_allowed_sent_at) = self.max_allowed_sent_at {
+            let elapsed = now.duration_since(max_allowed_sent_at);
+            let window_bytes_used = self.max_active - (self.max_allowed - self.retired);
+
+            // Same as `elapsed / rtt < window_bytes_used / max_active`
+            // without floating point division.
+            if elapsed.as_micros() * u128::from(self.max_active)
+                < rtt.as_micros() * u128::from(window_bytes_used)
```

Comment:
Is there a way to avoid `u128` math? It can be slow on some low-end ARM chips, and if we're executing this in the fast path we might want to avoid.

Reply:
Given that this happens at most 4 times per RTT, I'm OK with this being a tiny bit expensive.  If this were on a hot path, that would be different.

Reply:
With f351b880382616cfbf028d9f87e69e777f042522 algorithm uses `u64` instead of `u128`.